File: webhooks/verify.py
"""Verify an AntOps webhook body before processing it."""

import hashlib
import hmac
import os


def verify(raw_body: bytes, supplied_signature: str) -> bool:
    secret = os.environ["ANTOPS_WEBHOOK_SECRET"].encode()
    expected = hmac.new(secret, raw_body, hashlib.sha256).hexdigest()
    return hmac.compare_digest(expected, supplied_signature)

File: webhooks/verify.mjs
import { createHmac, timingSafeEqual } from "node:crypto";

export function verifyAntOpsWebhook(rawBody, suppliedSignature) {
  const expected = createHmac("sha256", process.env.ANTOPS_WEBHOOK_SECRET)
    .update(rawBody)
    .digest("hex");
  return timingSafeEqual(Buffer.from(expected), Buffer.from(suppliedSignature));
}
